device_doctor/lib/src/ios_device.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show LineSplitter;
+
+import 'package:meta/meta.dart';
+
+import 'device.dart';
+import 'process_helper.dart';
+
+/// IOS implementation of [DeviceDiscovery].
+///
+/// Discovers available ios devices and chooses one to work with.
+class IosDeviceDiscovery implements DeviceDiscovery {
+  factory IosDeviceDiscovery() {
+    return _instance ??= IosDeviceDiscovery._();
+  }
+
+  IosDeviceDiscovery._();
+
+  @visibleForTesting
+  IosDeviceDiscovery.testing();
+
+  static IosDeviceDiscovery _instance;
+
+  @override
+  Future<List<Device>> discoverDevices({int retriesDelayMs = 10000}) async {
+    final List<String> iosDeviceIds = LineSplitter.split(await deviceListOutput()).toList();
```

Comment:
you don't need to convert it to a List to map it, consider:

```
LineSplitter.split(await deviceListOutput())
  .map((String id) => IosDevice(deviceId: id))
  .toList()
```

Reply:
Thanks. Updated.